keps/sig-scheduling/4815-dra-partitionable-devices/README.md

```diff
@@ -252,222 +252,461 @@ non-goals of this KEP.
-	// their name. All others must be listed with their domain prefix.
-	//
-	// Conflicting attributes from those provided via other mixins are
-	// overwritten by the ones provided here.
-	//
-	// The maximum number of attributes and capacities combined is 32.
-	//
-	// +optional
-	Attributes map[QualifiedName]DeviceAttribute `json:"attributes,omitempty"`
-
-	// Capacity defines the set of capacities for this mixin.
-	// The name of each capacity must be unique in that set.
-	//
-	// To ensure this uniqueness, capacities defined by the vendor
-	// must be listed without the driver name as domain prefix in
-	// their name. All others must be listed with their domain prefix.
-	//
-	// Conflicting capacities from those provided via other mixins are
-	// overwritten by the ones provided here.
-	//
-	// The maximum number of attributes and capacities combined is 32.
-	//
-	// +optional
-	Capacity map[QualifiedName]DeviceCapacity `json:"capacity,omitempty"`
+  // Attributes defines the set of attributes for this mixin.
+  // The name of each attribute must be unique in that set.
+  //
+  // To ensure this uniqueness, attributes defined by the vendor
+  // must be listed without the driver name as domain prefix in
+  // their name. All others must be listed with their domain prefix.
+  //
+  // Conflicting attributes from those provided via other mixins are
+  // overwritten by the ones provided here.
+  //
+  // The maximum number of attributes and capacities combined is 32.
+  //
+  // +optional
+  Attributes map[QualifiedName]DeviceAttribute
+
+  // Capacity defines the set of capacities for this mixin.
+  // The name of each capacity must be unique in that set.
+  //
+  // To ensure this uniqueness, capacities defined by the vendor
+  // must be listed without the driver name as domain prefix in
+  // their name. All others must be listed with their domain prefix.
+  //
+  // Conflicting capacities from those provided via other mixins are
+  // overwritten by the ones provided here.
+  //
+  // The maximum number of attributes and capacities combined is 32.
+  //
+  // +optional
+  Capacity map[QualifiedName]DeviceCapacity
+}
+
+// DeviceCapacityConsumptionMixin defines a mixin that composite
+// devices can include to adopt the consuption capacity defined in
+// the mixin.
+type DeviceCapacityConsumptionMixin struct {
+  // Name is a unique identifier among all device capacity consumption 
+  // mixins in the ResourceSlice. It must be a DNS label.
+  //
+  // +required
+  Name string
+
+  // CapacityConsumption defines a set of capacities
+  // that a device will consume from a capacity pool.
+  //
+  // The capacity pool is not specified here but is determined
+  // from the context in which the DeviceCapacityConsumptionMixin
+  // is referenced from the device.
+  // 
+  // The maximum number of capacities is 32
+  //
+  // +required
+  CapacityConsumption map[QualifiedName]DeviceCapacity
```

Comment:
I think we can just call this `capacity` -- its clear from where its used, what the context is.

Reply:
Yeah, agree. I think the name of this field should be the same in both the mixin and the device (i.e. the `DeviceCapacityConsumption` struct) should match, so I've updated both.